src/libsyntax/codemap.rs

```diff
@@ -807,6 +807,70 @@ impl CodeMap {
                         hi.col.to_usize() + 1)).to_string()
     }
 
+    pub fn span_to_expanded_string(&self, sp: Span) -> String {
+        let src_str = self.span_to_string(sp);
+        let src_snip = self.span_to_snippet(sp).unwrap_or(
+            "Snippet unavailable (span malformed or source missing)".to_string());
+        if sp.expn_id == NO_EXPANSION || sp.expn_id == COMMAND_LINE_EXPN {
+            return format!("{}\n{}", src_snip, src_str);
+        }
+
+        let mut strings = vec![src_str.into_boxed_str(), src_snip.into_boxed_str()];
```

Comment:
Needed to use Box<str> inside the while block so the strings I made in it could survive beyond that scope, boxed those two so they'd match the type of the vector.